### DISCUSSION TITLE
@types/jsonwebtoken: should all non-standard claims in JwtPayload type be unknown?

### DISCUSSION THREAD

**Discussion by JasonShin** [DefinitelyTyped/DefinitelyTyped#54740] [Request a new @types package]
https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/jsonwebtoken/index.d.ts#L127

In here, I'm thinking it can be `[key: string]: unknown` and the user should write a type guard for each custom claim and transform the payload into their custom JwtPayload type.

**Comment by infodusha** [comment]
I completely like `unknown` approach.
Also, i would propose making functions that use JwtPayload generics, for example:
```typescript
export function verify<T extends JwtPayload = JwtPayload>(token: string, secretOrPublicKey: Secret, options?: VerifyOptions): Jwt<T> | T | string;
```
